test/python/pulse/test_instructions.py

```diff
@@ -121,29 +238,111 @@ def test_operator_delay(self):
         self.assertEqual(op_delay, op_identity)
 
 
+class TestFrameInstruction(QiskitTestCase):
+    """FrameInstruction common tests"""
+
+    def test_frame_instruction_input_validation(self):
+        """FrameInstructions share the input validation code.
+        Therefore, these tests use SetPhase as a representative example"""
+        channel = channels.DriveChannel(0)
+        target = Qubit(0)
+        frame = QubitFrame(0)
+        mixed_frame = MixedFrame(Qubit(0), QubitFrame(0))
+
+        # Invalid Combinations
+        with self.assertRaises(exceptions.PulseError):
+            instructions.SetPhase(0.5, target=target)
```

Comment:
This test is tied to the implementation of `SetPhase`. You need to define a clean test class within the test case which is independent of existing subclasses to be tested. Something like

```python
class TestInstruction(FrameInstruction):
    def __init__(
        *,
        frame: Frame = None,
        target: PulseTarget = None,
        mixed_frame: MixedFrame = None,
        channel: PulseChannel = None,
    ):
        ...
```